runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,452 @@
+end
+
+function M.idl(bufnr)
+  for _, line in ipairs(get_lines(bufnr, 1, 50)) do
+    -- Always ignore case
+    line = line:lower()
+    if line:find('^%s*import%s+"unknwn"%.idl') or line:find('^%s*import%s+"objidl"%.idl') then
+      vim.bo[bufnr].filetype = "msidl"
+      return
+    end
+  end
+  vim.bo[bufnr].filetype = "idl"
+end
+
+function M.inc(path, bufnr)
+
+end
+
+function M.inp(bufnr)
+  if get_lines(bufnr, 1):find("^%*") then
+    vim.bo[bufnr].filetype = "abaqus"
+  else
+    for _, line in ipairs(get_lines(bufnr, 1, 500)) do
+      if line:find("^header surface data") then
+        vim.bo[bufnr].filetype = "trasys"
+        return
+      end
+    end
+  end
+end
+
+function M.is_rapid(path, bufnr)
+
+end
+
+function M.lpc(path, bufnr)
+
+end
+
+function M.lprolog(path, bufnr)
+
+end
+
+function M.m(path, bufnr)
+
+end
+
+-- Rely on the file to start with a comment.
+-- MS message text files use ';', Sendmail files use '#' or 'dnl'
+function M.mc(bufnr)
+  for _, line in ipairs(get_lines(bufnr, 1, 20)) do
+    if line:find("^%s*#") or line:find("^%s*[dD][nN][lL]") then
+      -- Sendmail .mc file
+      vim.bo[bufnr].filetype = "m4"
+      return
+    elseif line:find("^%s*;") then
+      vim.bo[bufnr].filetype = "msmessages"
+      return
+    end
+  end
+  -- Default: Sendmail .mc file
+  vim.bo[bufnr].filetype = "m4"
+end
+
+function M.mm(path, bufnr)
+
+end
+
+function M.mms(bufnr)
+  for _, line in ipairs(get_lines(bufnr, 1, 20)) do
+    if line:find("^%s*%%") or line:find("^%s*//") or line:find("^%*") then
+      vim.bo[bufnr].filetype = "mmix"
+      return
+    elseif line:find("^%s*#") then
+      vim.bo[bufnr].filetype = "make"
+      return
+    end
+  end
+  vim.bo[bufnr].filetype = "mmix"
+end
+
+function M.mod(path, bufnr)
+
+end
+
+-- This function checks if one of the first five lines start with a dot. In
+-- that case it is probably an nroff file: 'filetype' is set and 1 is returned.
+function M.nroff(bufnr)
+  for _, line in ipairs(get_lines(bufnr, 1, 5)) do
+    if line:find("^%.") then
+      vim.bo[bufnr].filetype = "nroff"
+      return 1
+    end
+  end
+  return 0
+end
+
+function M.perl(path, bufnr)
+
+end
+
+function M.pl(path, bufnr)
+
+end
+
+function M.pp(path, bufnr)
+
+end
+
+function M.prg(path, bufnr)
+
+end
+
+function M.progress_asm(path, bufnr)
+
+end
+
+function M.progress_cweb(bufnr)
+  if vim.g.filetype_w then
+    vim.bo[bufnr].filetype = vim.g.filetype_w
+  else
+    if get_lines(bufnr, 1) == "&ANALYZE" or get_lines(bufnr, 3) == "&GLOBAL-DEFINE" then
+      vim.bo[bufnr].filetype = "progress"
+    else
+      vim.bo[bufnr].filetype = "cweb"
+    end
+  end
+end
+
+function M.progress_pascal(path, bufnr)
+
+end
+
+function M.proto(path, bufnr)
+
+end
+
+function M.r(bufnr)
+  local lines = get_lines(bufnr, 1, 50)
+  -- TODO: \< / \> which match the beginning / end of a word
+  -- Rebol is easy to recognize, check for that first
+  if table.concat(lines):find("[rR][eE][bB][oO][lL]") then
```

Comment:
Might be easier/simpler to just lower case it first.

```suggestion
  if table.concat(lines):lower():find("rebel") then
```